nomad/client_rpc.go

```diff
@@ -212,6 +212,40 @@ func (s *Server) serverWithNodeConn(nodeID, region string) (*serverParts, error)
 	return mostRecentServer, nil
 }
 
+// forwardClientRPC forwards the RPC specified by method to the node specified
+// by nodeID. Must be done after region forwarding, metrics, and permissions
+// checks.
+//
+// This is a wrapper method for getNodeForRpc, getNodeConn, etc that Client
+// RPCs which only need Servers to forward requests can use.
+func (s *Server) forwardClientRPC(method, nodeID string, args, reply any) error {
```

Comment:
Nice, we have this pattern in a few places in `client_csi_endpoint.go` where we could use this too.